hedgehog/hedgehog.cabal

```diff
@@ -134,3 +137,29 @@ test-suite test
     , semigroups                      >= 0.16       && < 0.19
     , text                            >= 1.1        && < 1.3
     , transformers                    >= 0.3        && < 0.6
+
+test-suite internals
+  type:
+    exitcode-stdio-1.0
+
+  main-is:
+    internals.hs
+
+  ghc-options:
+    -Wall -threaded -O2 -fno-warn-orphans
+
+  hs-source-dirs:
+    test
+
+  build-depends:
+      hedgehog
+    , QuickCheck
+    , base
+    , checkers
+    , tasty
+    , tasty-expected-failure
+    , tasty-quickcheck
+    , transformers
```

Comment:
I've been thinking about this for a while. Isn't it kind of oxymoron to use QC to test Hedgehog internals?

Perhaps, something like [this](https://github.com/hedgehogqa/haskell-hedgehog/pull/173/files#r220116453) allows us to mix and match properties as well as deterministic tests, by just using Hedgehog.

---

Feeling a bit worried about all the implicit QC arbitrary-magic. @jystic shows how a bug slipped in [bos/text](https://github.com/bos/text) library by using default arbitraries. It's on [this](https://youtu.be/AIv_9T0xKEo?t=710) video at 11:50 (up to 13:09).

/cc @jystic @thumphries

Reply:
I'm definitely torn on this idea myself, but I took the plunge and self-tested in scala-hedgehog:

https://github.com/hedgehogqa/scala-hedgehog/pull/16

I haven't worked out a way that will increase the level of confidence I have that the tests are actually running/working though.

Reply:
>I haven't worked out a way that will increase the level of confidence I have that the tests are actually running/working though.

That could very well be the case with any other testing tool we use; its runner might not be picking up our tests, etc.